server/konnectors/facebook_events.js

```diff
@@ -0,0 +1,242 @@
+'use strict';
+
+const async = require('async');
+const request = require('request');
+const moment = require('moment');
+const toQueryString = require('querystring').stringify;
+
+const baseKonnector = require('../lib/base_konnector');
+const localization = require('../lib/localization_manager');
+
+const Event = require('../models/event');
+
+const API_ROOT = 'https://graph.facebook.com/v2.6/';
+
+const appId = '991700800927312';
+const appSecret = 'a04e8cf918a382ea0b19cf1b6fbc2506';
+
+const scope = 'user_events';
+
+
+/*
+ * The goal of this connector is to fetch event from facebook and store them
+ * in the Cozy
+ */
+const connector = module.exports = baseKonnector.createNew({
+  name: 'Facebook Events',
+  slug: 'facebook_events',
+  customView: `<a href=${getOAuthProxyUrl()} target="_blank" >Connect</a>`,
+
+  fields: {
+    accessToken: 'text',
+    calendar: 'text',
+  },
+
+  models: [Event],
+
+  fetchOperations: [
+    updateToken,
+    saveTokenInKonnector,
+    downloadData,
+    parseData,
+    saveEvents,
+    buildNotifContent,
+  ],
+
+});
+
+function getOAuthProxyUrl() {
+  const baseUri = 'https://jacquarg.github.io/proxy_redirect/facebook_events/';
+  const params = {
+    appId,
+    scope,
+    redirect: 'display',
+  };
+
+  return `${baseUri}?${toQueryString(params)}`;
+}
+
+
+function updateToken(requiredFields, entries, data, next) {
+  connector.logger.info('Update facebook token');
+
+  const params = {
+    grant_type: 'fb_exchange_token',
+    client_id: appId,
+    client_secret: appSecret,
+    fb_exchange_token: requiredFields.accessToken,
+  };
+
+  request.get(`${API_ROOT}oauth/access_token?${toQueryString(params)}`,
+    (err, res, body) => {
+      if (err) {
+        connector.logger.error(`Update token failed: ${err.msg}`);
+        // TODO : notification token is broken, "reconnect"
+      } else {
+        data.accessToken = JSON.parse(body).access_token;
+      }
+      next(err);
+    });
+}
+
+// Save konnector's fieldValues during fetch process.
+function saveTokenInKonnector(requiredFields, entries, data, callback) {
+  connector.logger.info('Save refreshed token.');
+
+  const Konnector = require('../models/konnector');
```

Comment:
Konnector is empty ( `{}` ) if require is moved to the top. The reason is : 
1. [Konnector requires konnector_hash](https://github.com/cozy-labs/konnectors/blob/master/server/models/konnector.coffee#L3)
2. [Which load each konnectors modules](https://github.com/cozy-labs/konnectors/blob/master/server/lib/konnector_hash.coffee#L22-L26) 
So, if we requires Konnector in the body of a konnector, we get the Konnector at [this step](https://github.com/cozy-labs/konnectors/blob/master/server/models/konnector.coffee#L3), which is `{}`